user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/MeasurementMain.java

```diff
@@ -114,15 +126,78 @@ private void initSearchFieldAndButtonBar() {
     measurementSearchField.setValueChangeMode(ValueChangeMode.LAZY);
     measurementSearchField.addValueChangeListener(
         event -> measurementDetailsComponent.setSearchedMeasurementValue((event.getValue())));
+    Button downloadButton = new Button("Download Metadata");
+    downloadButton.addClickListener(event -> downloadMetadata());
     Button registerMeasurementButton = new Button("Register Measurements");
     registerMeasurementButton.addClassName("primary");
     registerMeasurementButton.addClickListener(
         event -> openRegisterMeasurementDialog());
-    Span buttonAndField = new Span(measurementSearchField, registerMeasurementButton);
+
+    Button editButton = new Button("Edit");
+    editButton.addClickListener(event -> openEditMeasurementDialog());
+    Span buttonAndField = new Span(measurementSearchField, downloadButton, editButton,
+        registerMeasurementButton);
     buttonAndField.addClassName("buttonAndField");
     content.add(buttonAndField);
   }
 
+  private Dialog setupDialog(MeasurementMetadataUploadDialog dialog, boolean editMode) {
+    dialog.addCancelListener(cancelEvent -> cancelEvent.getSource().close());
+    dialog.addConfirmListener(confirmEvent -> {
+      var uploads = confirmEvent.uploads();
+      boolean allSuccessfull = true;
+      for (var upload : uploads) {
+        try {
+          if (editMode) {
+            measurementService.updateMultiple(upload.measurementMetadata(),
+                context.projectId().orElseThrow());
+          } else {
+            measurementService.registerMultiple(upload.measurementMetadata(),
+                context.projectId().orElseThrow());
+          }
+        } catch (MeasurementRegistrationException measurementRegistrationException) {
+          allSuccessfull = false;
+          String errorMessage = switch (measurementRegistrationException.reason()) {
+            case FAILED, SUCCESSFUL -> "Registration failed. Please try again.";
```

Comment:
SUCCESSFUL should not lead to registration failed?

Reply:
Thank you, indeed successful is wrong, since it is not an error code at all. I removed it from the enum.